core/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -478,6 +483,12 @@ public static IndexTemplateMetaData fromXContent(XContentParser parser, String t
                                 }
                             }
                         }
+                    } else if ("template".equals(currentFieldName)) {
+                        List<String> templates = new ArrayList<>();
```

Comment:
This code is at a [local minima](https://en.wikipedia.org/wiki/Maxima_and_minima). It'd be more work to convert it to `ObjectParser` but every time we add more stuff here it makes it yet more work to do the conversion. This change is fine, but maybe, if you have time, you could consider converting this to use `ObjectParser` in a followup.